manual: Small style+grammar fixes to front matter and foreword

pmetzger opened the pull request:
I'm going through the OCaml manual, making small fixes to the style and grammar.
Mostly I'm following the rules of the Chicago Manual of Style, with some reference to other style manuals.
I'll be doing roughly one PR per chapter.

manual/manual/allfiles.etex:
    Change to put the fact that this is the manual in the main title,
    with the version in the subtitle.
    Capitalize title and chapter names per common editorial style
    guides like CMS, APA, MLA, etc.

manual/manual/foreword.etex:
    Capitalize title and chapter names per common editorial style
    guides like CMS, APA, MLA, etc.
    Small grammar fixes.
    The ocaml.org web site is for OCaml, not Caml, adjust accordingly.

lpw25 commented:
Most of the changes are capitalising titles. This  which is basically just changing from UK style to US style. Why would we prefer the latter? Changes that actually fix grammar are good, but I see no need to make changes that are clearly a matter of taste.

xavierleroy commented:
I hate capitalized titles with a passion.  They look like mock German.  It's not just a US vs UK thing as @lpw25 writes: looking at US newspapers, the New York Times and the Wall Street Journal still use capitalized headlines, but most other newspapers seem to use normal headlines.

pmetzger commented:
This capitalization style is what you find in the Chicago Manual of Style, the AP manual, the Modern Language Association style guide, The Elements of Style by Strunk and White, and pretty much every other guide. When I pull random books off the shelf, the words in the title are capitalized, as are chapter names, except in cases where chapter names are full sentences, when different rules apply.

> This which is basically just changing from UK style to US style. Why would we prefer the latter?

The short answer is, so far as I can tell after checking, this is also the UK style.

I'm not British so I'm not personally aware of what UK style might be, but the books published in the UK that I have all seem to follow the same style. The Oxford Manual of Style appears to indicate that in the UK, newspaper headlines are not title capitalized as they follow sentence rules, but book titles, subtitles, etc., are indeed capitalized by the same style as the US. Oxford University's style guide (not the same as the Oxford Manual of Style, and note the UK spelling) explicitly states:

> Capitalise the first word of the title, and all words within the title except articles (a/an/the), prepositions (to/on/for etc) and conjunctions (but/and/or etc). 

I recognize that to a French reader this might seem alien, but I assure you that to a native reader of English, the capitalization of the existing book title and section names etc. is very alien and stands out immediately. You can hate this with a passion, but note that to an educated American reader (and likely to an educated British reader, though I am not British and can't speak personally to it), the existing style looks horrible because it violates a lifetime of expectations.

pmetzger commented:
I've now consulted a number of UK published books on my shelf. I can provide photographs of the title, subtitle, chapter names, etc. They all follow the same rule that I'm familiar with from my education in the United States. The first and last words are always capitalized, as are all other words except for short prepositions and articles. I will, for example, post photographs from a Penguin UK edition of the complete works of Lewis Carroll if it is demanded. (And again, newspaper headline styles differ between the US and UK, but as the OCaml manual is not a set of newspaper stories, this does not matter.)

pmetzger commented:
BTW, just a side note: all rules of grammar and style are fundamentally arbitrary, just as all the rules of the OCaml syntax and grammar are fundamentally arbitrary. However, if you ignore the rules of OCaml syntax and grammar, the compiler will have a hard time intuiting what you meant, and if you ignore OCaml style rules, your program will be harder to maintain and less readable. If you ignore the style and grammar rules of a natural language, it isn't quite so bad, as readers will still intuit what you meant, but it will often slow them down while reading, and it will absolutely make your document seem less professional.

I personally find the gender rules of French and German entirely deranged and arbitrary, but if one is communicating in those languages, one must follow those rules even though they often assign sexes to objects that are clearly inanimate and genderless, and even sometimes assign incorrect sexes to things that are animate and gendered. If one violates these rules, a native speaker will absolutely still understand a document, but they'll also be taken aback by the lack of adherence to the rules, they will read the document more slowly because the errors will distract them, and they will probably judge the document more harshly even if the content is fundamentally understandable.

dra27 commented:
> I'll be doing roughly one PR per chapter

Is this really necessary - it would be nicer to review the set of changes once proposed all at once?

I'm also slightly bemused at splitting out related work into multiple PRs, yet merging unrelated work in single commits - the `*.svg` correction is independent from this and certainly worthy of its own commit. Please could that at least be split to a separate commit, if not even to its own PR?

FWIW I think there's an apples and oranges comparison going on - chapter titles and headings are not headlines. I find both "Exciting Thing Happens In Some City In The World" odd in a newspaper and "The OCaml language" odd in a table of contents.

pmetzger commented:
@dra27:

> Is this really necessary - it would be nicer to review the set of changes once proposed all at once?

I could do that, but on a chat channel (the "docs" channel of the OCaml discord) there was a brief discussion on it, and a couple of people suggested that one per chapter would reduce the complexity of the review process. If the consensus is that I should do the whole thing at once, I can do that, but it will end up being a lot of content to review in one shot.

> I'm also slightly bemused at splitting out related work into multiple PRs, yet merging unrelated work in single commits - the *.svg correction is independent from this and certainly worthy of its own commit.

Yah, I feel a bit weird about that one too. I will separate it out.

> FWIW I think there's an apples and oranges comparison going on - chapter titles and headings are not headlines. I find both "Exciting Thing Happens In Some City In The World" odd in a newspaper and "The OCaml language" odd in a table of contents.

My changes are all to the latter sort of thing, of course. Although the UK has a distinct headline style from the US, the rules for things like book titles and chapter names are the same. (There's more of a bias in the Oxford Manual towards the use of sentence-style capitalization if chapter names are complete sentences, but that's not the case in the OCaml manual so I went for the more usual style choice.)

pmetzger commented:
Oh, one other question: I really don't feel this stuff is worthy of a "Changes" entry. What do I do (if anything) to silence Travis on that?

pmetzger committed:
manual: Small style and grammar fixes in front matter and foreword

manual/manual/allfiles.etex:
    Change to put the fact that this is the manual in the main title,
    with the version in the subtitle.
    Capitalize title and chapter names per common editorial style
    guides like CMS, APA, MLA, etc.

manual/manual/foreword.etex:
    Capitalize title and chapter names per common editorial style
    guides like CMS, APA, MLA, etc.
    Small grammar fixes.

pmetzger commented:
@dra27 The .svg cleanup is now be part of a distinct pull request: https://github.com/ocaml/ocaml/pull/1889

lpw25 commented:
Book titles generally use title case, but UK publications do not use title case for headlines, article titles, chapter headings, section headings, etc. For example, note the headings in the Economist and BBC style guides:

  https://bordeure.files.wordpress.com/2008/11/the-economist-style-guide.pdf

  http://www2.media.uoa.gr/lectures/linguistic_archives/academic_papers0506/notes/stylesheets_3.pdf

Using sentence case for these things is in the minority in US publications, but is still present in plenty of them (e.g. Washington Post, LA Times).

So this is not some universally accepted standard, it is a matter of taste, and there is no need to change it. (Personally, I agree with Xavier that they look bad, but that isn't the main point, the point is not to make unnecessary changes to suit your personal taste).

lpw25 commented:
>  "The OCaml language" odd in a table of contents.

I suggest you grab a textbook published by your own university's press or one from the other place. The two closest to where I'm sitting ("Quantum Computation and Quantum Information", Cambridge University Press, "Probability and Random Processes", Oxford University Press) both use sentence case for chapter titles and section headings.

pmetzger commented:
> Book titles generally use title case, but UK publications do not use title case for headlines, article titles, chapter headings, section headings, etc. For example, note the headings in the Economist and BBC style guides:

The BBC and The Economist are news organizations. Headlines and chapter titles do not follow the same rules. I will post numerous examples of actual UK publications if this desired. I will repeat the rule I've seen in UK style guides, which is to use capitals in chapter names in the usual manner _unless_ the chapter titles are full sentences. As the OCaml manual does not use full sentences for chapter titles, capitals are called for.

Perhaps it would be simply superior for me to start posting examples. I have an extensive collection of UK published books, all of which follow the rule as I've described it.

pmetzger commented:
Here is the table of contents of Roy Porter's "London: A Social History", published in the UK:

![img_8097](https://user-images.githubusercontent.com/1907413/42422504-669c8726-82b5-11e8-9c21-cddeb5e7c08b.JPG)

pmetzger commented:
Here is the table of contents from Anthony Burgess' "A Mouthful of Air", published by Stoddart in the UK. The edition was never made available for sale in the US:

![img_8098](https://user-images.githubusercontent.com/1907413/42422530-ece20a90-82b5-11e8-953d-31395e35cb5a.JPG)

pmetzger commented:
Here is the table of contents of Brian Aldiss' "Bury My Heart At W.H. Smith's", published by Hodder and Stoughton in the UK, and to my knowledge never released in this edition in the United States:

![img_8099](https://user-images.githubusercontent.com/1907413/42422564-7ea15cb0-82b6-11e8-822b-f1cea2a09aae.JPG)

thizanne commented:
Are you really going to post every single English book in the planet to prove that your proposed style is indeed universal, or do you just expect to start a table of contents photography battle ?

pmetzger commented:
> Are you really going to post every single English book in the planet to prove that your proposed style is indeed universal, or do you just expect to start a table of contents photography battle ?

If people wish to claim that they simply refuse to follow the usual style for works published in all English speaking countries, that's fine. I'm merely demonstrating that the style is both the one prescribed in the style manuals of both the United Kingdom and United States and is universally followed through the use of direct evidence. @lpw25 claimed otherwise, and I'm demonstrating that he's incorrect. (I agree with him that chapter titles that are _sentences_ follow sentence casing rules for publications in the United Kingdom, but the OCaml manual has no such chapter titles.)

pmetzger commented:
More examples on request, by the way. I've checked "The Penguin Complete Lewis Carroll", Julian Barnes' "A History of the World in 10 1/2 Chapters" (published in the UK by Picador) and a dozen more. The only exception I've found so far was no exception: it was a history of Hong Kong published in the UK in which chapter titles were complete sentences (thus falling under the alternate rule.)

xavierleroy commented:
Who would have thought that such a trivial matter would generate such a discussion...  I wish similar attention was paid to the great many more serious issues that OCaml has.  I also resent being dragged in such a discussion while I'd rather spend my time otherwise.

This said, my opinion hasn't changed and I'm in strong agreement with @lpw25, no matter how many scans @pmetzger posts, so please stop. 

To state things more clearly:

I am the original author of the parts of the OCaml manual that are covered by this PR.  

I chose to use uncapitalized titles not by French ignorance, but being fully aware of Strunk&White, Chicago manual of style, and the like.  (Back in those days I already fought against copy editors for various journals.)  

I chose to use uncapitalized titles because I know they are used by a great many English publications and because I find them less fussy and more modern than the capitalized versions.

In 20+ years I got many criticisms about the OCaml manual, most of them justified, but the capitalization style was mentioned only once, by a French person who thought the Chicago manual of style is normative.

I stand by those stylistic choices and will not be forced into changing them.

@pmetzger : either you submit PRs about content, not style, or I'll close them.

pmetzger commented:
Fine. I'll abandon my PR then.
